### ISSUE TITLE
xb.Op.scalar maps unsigned int to torch signed int

### ISSUE THREAD

**Issue by tengyifei** [pytorch/xla#8798]
torch supports unsigned int types so this is probably a bug.

**Comment by iwknow** [comment]
is there an instruction to reproduce the issue? if it is easy to reproduce, i can take a look.

**Comment by rpsilva-aws** [comment]
I don't have an exact repro at the moment, but I can write down on how we could go about it.
The issue refers to `_XLA_PT_TYPE_MAP`.

The only place where this exists today is:
```
  @classmethod
  def scalar(cls, builder, value, dtype=None):
    return mkleaf(
        'Constant',
        builder,
        value=torch.tensor(value, dtype=cls.to_torch_type(dtype)))
```

Hence, if you can use the XLA builder to generate a computation that derives the type of a torch.uint32, but generates an XLA S32 type - we could use that to trigger either a lowering error or an accuracy issue (e.g. overflow).

You can start by looking at `test_operations.py` to mimic/repro the behavior. If I have time, I can generate one.

**Comment by rpsilva-aws** [comment]
Correct. Ideally coming up with an actual test case that would've failed or caused incorrect results with the current implementation, that would've now correctly succeed with your change. This would be the main scope of the item.

**Comment by iwknow** [comment]
ok. i will take a look
